radix-engine/src/engine/modules/execution_trace.rs

```diff
@@ -164,11 +181,22 @@ impl ExecutionTrace {
             .or_insert((resource_address.clone(), Decimal::zero()));
 
         vault_change.1 += amount;
-
-        Ok(())
     }
 
-    pub fn to_receipt(self) -> ExecutionTraceReceipt {
+    pub fn to_receipt(
+        mut self,
+        fee_payments: HashMap<VaultId, (ResourceAddress, Decimal)>,
```

Comment:
Only need a reference to the `HashMap` right?

Reply:
Looks like this is a map constructed for receipt generation only, so no big deal.